flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text("Sample App"),
+      ),
+      body: new ListView(children: _getListData()),
+    );
+  }
+
+  _getListData() {
+    List<Widget> widgets = new List();
+    for (int i = 0; i < 100; i++) {
+      widgets.add(new Padding(padding: new EdgeInsets.all(10.0), child: new Text("Row $i")));
+    }
+    return widgets;
+  }
+}
+{% endprettify %}
+
+**How do I know which list item is clicked on?**
+
+Unlike Android where a ListView offers an onItemClickListener, Flutter makes it 
+a lot easier by letting you just use the touch handling that the widgets you 
+passed in have.
+
+<!-- skip -->
+{% prettify dart %}
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(new SampleApp());
+}
+
+class SampleApp extends StatelessWidget {
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Sample App',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new SampleAppPage(),
+    );
+  }
+}
+
+class SampleAppPage extends StatefulWidget {
+  SampleAppPage({Key key}) : super(key: key);
+
+  @override
+  _SampleAppPageState createState() => new _SampleAppPageState();
+}
+
+class _SampleAppPageState extends State<SampleAppPage> {
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text("Sample App"),
+      ),
+      body: new ListView(children: _getListData()),
+    );
+  }
+
+  _getListData() {
+    List<Widget> widgets = new List();
+    for (int i = 0; i < 100; i++) {
+      widgets.add(new GestureDetector(
+        child: new Padding(padding: new EdgeInsets.all(10.0), child: new Text("Row $i")),
+        onTap: () {
+          print('row tapped');
+        },
+      ));
+    }
+    return widgets;
+  }
+}
+{% endprettify %}
+
+**How do I update ListView's dynamically?**
+
+This confuses a lot of people when coming from Android. On Android, you would 
+update the adapter and call notifyDataSetChanged. In Flutter if you were to 
+update the list of widgets inside a setState(), you would quickly see that your 
+data did not change visually. 
+
+This is because when setState is called, the Flutter rendering engine will go 
+through all the widgets to see if they have changed. When it gets to your 
+ListView it will do a `==operator` and see that the two ListViews are the same 
+and nothing has changed, hence no update to the data.
+
+The quick and dirty way around this is to assign your list variable a new 
+List() inside of setState, copy over all the old data to the new list. This is 
+inefficient a simple dirty way to achieve an update. 
```

Comment:
depends on the list implementation, and if you know the size of the list, it might not be too bad.

If we don't know how to quantify efficiency, I'd just say "this is a simple way to achieve the updated"

it's not dirty, IMO :)